modules/@apostrophecms/doc-type/index.js

```diff
@@ -1481,6 +1481,41 @@ module.exports = {
         });
 
         return doc;
+      },
+
+      composeFilters() {
+        self.filters = Object.keys(self.filters).map((key) => ({
+          name: key,
+          ...self.filters[key],
+          inputType: self.filters[key].inputType || 'select'
+        }));
+        // Add a null choice if not already added or set to `required`
+        self.filters.forEach((filter) => {
+          if (filter.choices) {
+            if (
+              !filter.required &&
+              filter.choices &&
+              !filter.choices.find((choice) => choice.value === null)
+            ) {
+              filter.def = null;
+              filter.choices.push({
+                value: null,
+                label: 'apostrophe:none'
+              });
+            }
+          } else {
+            // Dynamic choices from the REST API, but
+            // we need a label for "no opinion"
+            filter.nullLabel = 'Choose One';
+          }
+        });
+      },
+
+      composeColumns() {
+        self.columns = Object.keys(self.columns).map((key) => ({
+          name: key,
+          ...self.columns[key]
+        }));
```

Comment:
`composeFilters` and `composeColumns` factored out in doc-type because they are now used by `piece-type` and `page-type` both.

Reply:
I was startled but it makes sense